internal/x/xterminal/terminal_windows.go

```diff
@@ -85,29 +88,38 @@ func (s ResizeEvent) equal(s2 *ResizeEvent) bool {
 // ResizeEvents sends terminal resize events when the dimensions change.
 // Windows does not have a unix.SIGWINCH equivalent, so we poll the terminal size
 // at a fixed interval
-func ResizeEvents(ctx context.Context, termfd uintptr) chan ResizeEvent {
-	events := make(chan ResizeEvent)
-	ticker := time.NewTicker(time.Millisecond * 100)
+func ResizeEvents(ctx context.Context, termFD uintptr) chan ResizeEvent {
+	events := make(chan ResizeEvent, 1) // Use a buffered chan to avoid blocking.
```

Comment:
isn't blocking a symptom of a bug? if it works without buffering, I would leave it buffered?

Reply:
I'll update the comment to add more context about the reasoning.

We want to send an initial event to signal the current size of the window. I was planning to do this in a different PR as this one ended up being mostly about styling/cleaning.

At a higher level, the idea is that when dealing with the channel, the runtime notifies us via a non-blocking chan, so if we are consuming it on an unbuffered chan, we lose the event.
In this specific case, we are dealing with windows and it is not actually a signal, but it maps directly with how it works on unix.